source/octf/trace/parser/ParsedIoTraceEventHandler.cpp

```diff
@@ -662,15 +664,18 @@ class ParsedIoTraceEventHandler::FileSystemViewer : public IFileSystemViewer {
     }
 
 private:
-    bool getPath(uint64_t id, std::string &path) const {
+    bool getPath(uint64_t id, std::string &path, uint64_t &len) const {
         FileId fid(m_partId, id);
 
         auto iter = m_fileInfo.find(fid);
         if (iter != m_fileInfo.end()) {
             const auto &info = iter->second;
-
+            len += info.name.length();
+            if (len > PATH_MAX) {
+                throw Exception("Exceeded maximum path length limit");
```

Comment:
When throwing exception stop parsing at all. But this situation rarely may appear and we would like to continue parsing, but for such particular case won't have a path. Please throw exception (but custom) and catch it at the top of this procedure and return empty path.